library/core/src/num/dec2flt/float.rs

```diff
@@ -0,0 +1,208 @@
+//! Helper trait for generic float types.
+
+use crate::fmt::{Debug, Display, LowerExp};
+use crate::num::FpCategory;
+use crate::ops::{Add, Div, Mul, Neg};
+
+/// A helper trait to avoid duplicating basically all the conversion code for `f32` and `f64`.
+///
+/// See the parent module's doc comment for why this is necessary.
+///
+/// Should **never ever** be implemented for other types or be used outside the dec2flt module.
+#[doc(hidden)]
+pub trait RawFloat:
+    Sized
+    + Div<Output = Self>
+    + Neg<Output = Self>
+    + Mul<Output = Self>
+    + Add<Output = Self>
+    + LowerExp
+    + PartialEq
+    + PartialOrd
+    + Default
+    + Clone
+    + Copy
+    + Debug
+    + Display
```

Comment:
Are `Debug` / `Display` actually used in this PR? (I think all other bounds are)

Reply:
Probably not, they certainly aren't used for testing right now, although they might be in the future. The old `dec2flt` had `Debug` [implemented](https://github.com/rust-lang/rust/blob/71b8742bbcbed2cd908dbc031d6552d8b528c037/library/core/src/num/dec2flt/rawfp.rs#L48), so it might be worth keeping that... But `Display` should probably go.

Reply:
Display has been removed in the latest [commit](https://github.com/Alexhuszagh/rust/blob/870ed2163282b2e5fa99585e6c28c916e4861646/library/core/src/num/dec2flt/float.rs#L12-L25).